Packet++/header/NflogLayer.h

```diff
@@ -0,0 +1,272 @@
+		 * @param[in] recordRawData A pointer to the option raw data
+		 */
+		explicit NflogTLV(uint8_t* recordRawData)
+		{
+			assign(recordRawData);
+		}
+
+		/**
+		 * @return recordLen attribute in NflogTLVRawData
+		 */
+		size_t getTotalSize() const { return m_Data->recordLen; }
+
+		/**
+		 * Assign a pointer to the TLV record raw data (byte array)
+		 * @param[in] recordRawData A pointer to the TLV record raw data
+		 */
+		void assign(uint8_t* recordRawData)
+		{
+			if(recordRawData == nullptr)
+				m_Data = nullptr;
+			else
+			{
+				// to pass from some unknown paddings after tlv with type NFULA_PREFIX
+				while (*recordRawData == 0)
+					recordRawData += 1;
+				m_Data = (NflogTLVRawData*)recordRawData;
+			}
+		}
+
+		/**
+		 * @return True if the TLV record raw data is NULL, false otherwise
+		 */
+		bool isNull() const
+		{
+			return (m_Data == nullptr);
+		}
+
+		/**
+		 * @return The type field of the record (the 'T' in __Type__-Length-Value)
+		 */
+		uint16_t getType() const { return m_Data->recordType; }
+
+		/**
+		 * @return A pointer to the TLV record raw data byte stream
+		 */
+		uint8_t* getRecordBasePtr() const { return (uint8_t*)m_Data; }
+
+		/**
+		 * @return A pointer to the value of the record as byte array (the 'V' in Type-Length- __Value__)
+		 */
+		uint8_t* getValue() const { return m_Data->recordValue; }
+	};
+
+	/**
+	 * @class NflogLayer
+	 * Represents an NFLOG protocol layer
+	 */
+	class NflogLayer : public Layer
+	{
+	public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to ether_header)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		NflogLayer(uint8_t* data, size_t dataLen, Packet* packet) : Layer(data, dataLen, NULL, packet) { m_Protocol = NFLOG; }
+
+		~NflogLayer() {}
+
+		/**
+		 * Get a pointer to the Nflog header.
+		 * @return A pointer to the nflog_header
+		 */
+		nflog_header* getNflogHeader() const { return (nflog_header*)m_Data; }
+
+		/**
+		 * Get address family of the packet. e.g. 2 for ipv4 and 10 for ipv6
+		 * @return an unsigned char of address family
+		*/
+		uint8_t getFamily();
+
+        /**
+         * Get Version number inside packet header
+         * The version field is 0 for the current version of the pseudo-header
+         * @return an unsigned char for version
+        */
+        uint8_t getVersion();
+
+        /**
+         * Get Resource Id in packet header
+         * On one netlink socket it's possible to listen to several nflog groups; the resource ID is the nflog group for the packet
+        */
+        uint16_t getResourceId();
+
+		/**
+		 * get packet header of the packet as the tlv value of 1
+		 * @return pointer to nflog_packet_header
+		*/
+		nflog_packet_header* getPacketHeader();
```

Comment:
This method parses the first TLV, why is it important to provide a special method for it?

Reply:
I think in the beginning of the development, I got it wrong instead of nflog_header